resources/assets/lib/turbolinks-reload.ts

```diff
@@ -1,49 +1,36 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-const className = 'js-extra-script';
-
-interface ScriptList {
-  [key: string]: boolean;
-}
-
 export default class TurbolinksReload {
-  private loaded: ScriptList = {};
+  private loaded = new Set<string>();
+  private loading = new Map<string, JQuery.jqXHR<void>>();
 
   constructor() {
-    document.addEventListener('turbolinks:before-cache', this.cleanup);
+    $(document).on('turbolinks:before-cache', this.abortLoading);
   }
 
-  cleanup = () => {
-    $(`.${className}`).remove();
+  abortLoading = () => {
+    for (const xhr of this.loading.values()) {
+      xhr.abort();
```

Comment:
Probably should also should remove from the set after aborting

Reply:
should already be done by the always callback

Reply:
...and shouldn't be in loaded set if not done